servers/zms/src/main/java/com/yahoo/athenz/zms/store/AuthHistoryStoreFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The Athenz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.zms.store;
+
+public interface AuthHistoryStoreFactory {
+
+    /**
+     * Create and return a new AuthHistoryStore instance
+     * @return AuthHistoryStore instance
+     */
+    AuthHistoryStore create();
```

Comment:
we should consider what the interface would require for other implementations. Maybe at a minimum pass the private key store so the plugin can read username/password details if necessary.